sysutils/yabai/Portfile

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem              1.0
+PortGroup               github 1.0
+
+github.setup            koekeishiya yabai 3.3.0 v
+github.tarball_from     releases
+
+categories              sysutils
+platforms               darwin
+maintainers             {@thekevjames thekev.in:macports} openmaintainer
+license                 MIT
+description             A tiling window manager for macOS based on binary space partitioning
+long_description        yabai is a window management utility that is designed to work as an extension to the built-in window manager of macOS.
+distname                ${github.project}-v${github.version}
+
+supported_archs         x86_64
+checksums               rmd160  85d63f56aa1ef53b300ac3228236168305dbad50 \
+                        sha256  8ccb7ba4e3522fa6f52545a9d5ac0c5efb7b0720ed9c2986605022a01c9938ce \
+                        size    249829
+
+set yabai.plist         ${workpath}/org.macports.yabai.plist
+
```

Comment:
you need to make sure that the port is [UsingTheRightCompiler](https://trac.macports.org/wiki/UsingTheRightCompiler) and that will require a patch to the makefile.

So please add here `patchfiles-append patch-makefile.diff` and add a file with this name to the `files` directory with this content:

```
--- makefile.orig       2021-01-20 22:03:01.000000000 -0500
+++ makefile    2021-01-20 22:03:11.000000000 -0500
@@ -20,9 +20,9 @@
 install: clean-build $(BINS)

 $(OSAX_SRC): $(OSAX_PATH)/loader.m $(OSAX_PATH)/payload.m $(OSAX_PATH)/mach_bootstrap.c
-       clang $(OSAX_PATH)/loader.m -shared -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/loader -framework Foundation
-       clang $(OSAX_PATH)/payload.m -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/payload -framework Foundation -framework Carbon
-       clang $(OSAX_PATH)/mach_bootstrap.c -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/mach_bootstrap -framework Carbon -lpthread
+       ${CC} $(OSAX_PATH)/loader.m -shared -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/loader -framework Foundation
+       ${CC} $(OSAX_PATH)/payload.m -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/payload -framework Foundation -framework Carbon
+       ${CC} $(OSAX_PATH)/mach_bootstrap.c -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/mach_bootstrap -framework Carbon -lpthread
        xxd -i -a $(OSAX_PATH)/loader $(OSAX_PATH)/sa_loader.c
        xxd -i -a $(OSAX_PATH)/payload $(OSAX_PATH)/sa_payload.c
        xxd -i -a $(OSAX_PATH)/mach_bootstrap $(OSAX_PATH)/sa_mach_bootstrap.c
@@ -56,4 +56,4 @@

 $(BUILD_PATH)/yabai: $(YABAI_SRC)
        mkdir -p $(BUILD_PATH)
-       clang $^ $(BUILD_FLAGS) $(FRAMEWORK_PATH) $(FRAMEWORK) -o $@
+       ${CC} $^ $(BUILD_FLAGS) $(FRAMEWORK_PATH) $(FRAMEWORK) -o $@
```